agent/agent.go

```diff
@@ -550,11 +550,14 @@ func (a *Agent) Publisher(ctx context.Context, subscriptionID string) (exec.LogP
 func (a *Agent) nodeDescriptionWithHostname(ctx context.Context, tlsInfo *api.NodeTLSInfo) (*api.NodeDescription, error) {
 	desc, err := a.config.Executor.Describe(ctx)
 
-	// Override hostname and TLS info
+	// Override hostname
 	if desc != nil {
-		if a.config.Hostname != "" && desc != nil {
+		if a.config.Hostname != "" {
 			desc.Hostname = a.config.Hostname
 		}
+		if desc.Resources != nil && a.config.GenericResources != nil {
+			desc.Resources.Generic = a.config.GenericResources
```

Comment:
We would thread it through. For the docker executor, the memory and CPU resources are returned by `SystemInfo` call into the daemon backend:

https://github.com/docker/docker/blob/master/daemon/cluster/executor/container/executor.go#L107-L110

I think it would make sense to do the same for generic resource so they are handled in the same way.

For the `dockerapi` executor in this repository, there's a bit of a circular dependency: Docker would need to be updated to return generic resources from its `/info` API endpoint (which `Describe` is using via `e.client.Info`), but that would require Docker to support generic resources already.

As a temporary measure for `agent/exec/dockerapi`, we could keep the resource information you're putting on `Node`, and provide that to the executor in `NewExecutor`. `Describe` would return this resource info.

But once this is all integrated in Docker, I think the source of truth should be the `SystemInfo` call, not information passed through `Node`.